core/txbuilder/types.go

```diff
@@ -29,24 +29,24 @@ type Template struct {
 	sigHasher *bc.SigHasher
 }
 
-func (t *Template) Hash(idx int) bc.Hash {
+func (t *Template) Hash(idx uint32) bc.Hash {
 	if t.sigHasher == nil {
 		t.sigHasher = bc.NewSigHasher(t.Transaction)
 	}
-	return t.sigHasher.Hash(idx)
+	return t.sigHasher.Hash(int(idx))
```

Comment:
This is not a safe conversion. Maybe update `SigHasher.Hash` to take a `uint32`?